### PR TITLE
Testing/dashboard

### PR DESCRIPTION
**Pull request by LKessell**
### Expected Behavior
Install Cypress and set up testing file structure
Add tests for viewing main dashboard with posters, and internal server error message
Tests are passing

### Code Reviewers
@alexmkio 

### Professionalism Rubric
- README concisely communicates the team’s individual and joint learning goals, the evolution of the project, and team member reflections while using good formatting to enhance readability
- README links to all user GitHub profiles and any applicable repos/deployed sites
- Team uses a rebase workflow
- Git commits are atomic, with concise and precise descriptions of the change made
- PRs have full, consistent descriptions
- Team members do consistent, thorough, meaningful code reviews of PRs, which prompt updates and changes made to that PR before merging
- Evolution of the project (decisions made, etc) are fully and clearly documented in the git history and PRs
- When the project is run locally, the terminal shows no errors or warnings

### React Rubric
- A consistent, modular file structure is used
- A clear understanding of class components vs function components is demonstrated
- Only the data that a child component needs is passed down as props
- Logic is kept out of return statements; return statements are as readable as possible, only communicating what will be displayed
- Fetch calls have been refactored to be reusable for multiple queries
- Frontend data (state) always matches the backend data
- Data fetched from API is run through a cleaning function (which is defined in a separate utilities file)
- Implements excellent error handling if movie database server is down or fetch fails (this includes loading images as well as error messages on the frontend)

### Testing Rubric
- Team has successfully achieved 90%+ test coverage of all components
- All async functionality is stubbed and tested
- Async tests cover happy & sad paths
- All application views are tested
- All user flows are tested

### Routing Rubric
- Application has been refactored to use Router without leaving artifacts of the prior code (no odd workarounds remaining)
- Use of Router shows developer empathy (readability, maintainability)
- Application uses React Router components and does not manipulate the history object
- A 404 page handles undefined routes
- UX is excellent; routes are consistent and navigation is clear

### Mod1 Best Practices
- All functions strictly adhere to the [Single Responsibility Principle (SRP)](https://dev.to/skill_pathway/single-responsibility-principle-for-dummies-59gb) and are all 10 lines of code or less
- Uses logical operators instead of nesting conditional statements
- Conditionals are written in a way that demonstrates an understanding of [https://frontend.turing.io/lessons/module-1/js-truthy-falsy-expressions.html](truthy/falsy values). For example, the developer avoids using logic like <code class="language-plaintext highlighter-rouge">=== true</code> and <code class="language-plaintext highlighter-rouge">!== 0</code>.
- Functions are [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and observe [SRP](https://dev.to/skill_pathway/single-responsibility-principle-for-dummies-59gb) - and most are around 10 lines of code or less
- Most functions and variables are semantically and concisely named. Functions should start with a verb, array names should be plural, etc. Function and variable names describe their role in the program. Examples: The name of the data type should not ever be in a variable name (ex: “petArray”); the name itself should be clear enough to indicate the type of data it holds (ex: “allPets” or “pets”).
- An event handler should not have the word “handler” in the name (ex: “clickHandler”); the name should indicate the handler’s purpose (ex: “addNewPet”).
- Uses parameters and arguments to craft short and reusable functions. There are no parameters or arguments in functions that are unused.
- All console logs, debuggers and comments are removed from code before submitting.

### CODE REVIEW

File: cypress/fixtures/example.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (alexmkio) [inline cypress/fixtures/example.json]: can we get rid of this file?

File: cypress/integration/dashboard_spec.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          average_rating: 7,
+          release_date: '2020-08-20'
+        }
+      ]
+    });
+
+    cy.visit('http://localhost:3000')
+      .get('h1').contains('Rancid Tomatillos')
+      .get('h2').contains('Featured Films')
+      .get('img[id="694919"]').should('be.visible')
```
Comment (alexmkio) [inline cypress/integration/dashboard_spec.js:40]: is bracket notation necessary here?

File: cypress/support/commands.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// ***********************************************
+// This example commands.js shows you how to
```
Comment (alexmkio) [inline cypress/support/commands.js]: can we get rid of this file?

File: cypress.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (alexmkio) [inline cypress.json:1]: can we get rid of this file?

File: cypress/support/index.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+// ***********************************************************
```
Comment (alexmkio) [inline cypress/support/index.js]: can we get rid of this file?

File: cypress/integration/dashboard_spec.js (hunk 2 above)
Comment (LKessell) [reply cypress/integration/dashboard_spec.js:40]: Yup, this is cypress syntax targeting the image with the specified id

File: cypress.json (hunk 4 above)
Comment (LKessell) [reply cypress.json:1]: Going to keep this one for now, in case we add any special configurations!

File: cypress/fixtures/example.json (hunk 1 above)
Comment (LKessell) [reply cypress/fixtures/example.json]: Removed

File: cypress/support/commands.js (hunk 3 above)
Comment (LKessell) [reply cypress/support/commands.js]: Removed

File: cypress/support/index.js (hunk 5 above)
Comment (LKessell) [reply cypress/support/index.js]: Removed

**Pull request merged**